app/services/S3.scala

```diff
@@ -0,0 +1,94 @@
+package services
+
+import com.amazonaws.services.s3.{AmazonS3, AmazonS3ClientBuilder}
+import com.typesafe.scalalogging.StrictLogging
+import io.circe.{Decoder, Encoder, Json, Printer}
+import io.circe.parser.decode
+import io.circe.syntax._
+import services.S3Client.RawVersionedS3Data
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+case class VersionedS3Data[T](value: T, version: String)
+
+trait S3Client {
+  def get(bucket: String, key: String)(implicit ec: ExecutionContext): Future[Either[String,RawVersionedS3Data]]
+  def put(bucket: String, key: String, data: RawVersionedS3Data)(implicit ec: ExecutionContext): Future[Either[String,RawVersionedS3Data]]
+}
+
+object S3Client {
+  type RawVersionedS3Data = VersionedS3Data[String]
+}
+
+object S3 extends S3Client with StrictLogging {
+
+  private val s3Client: AmazonS3 = AmazonS3ClientBuilder
+    .standard()
+    .withRegion(Aws.region)
+    .withCredentials(Aws.credentialsProvider)
+    .build()
+
+  def get(bucket: String, key: String)(implicit ec: ExecutionContext): Future[Either[String,RawVersionedS3Data]] = Future {
+    try {
+      val s3Object = s3Client.getObject(bucket, key)
+
+      val version = s3Object.getObjectMetadata.getVersionId
+
+      val stream = s3Object.getObjectContent
+      val raw: String = scala.io.Source.fromInputStream(stream).mkString
+      stream.close()
+
+      Right[String,RawVersionedS3Data](VersionedS3Data(raw, version))
+
+    } catch {
+      case NonFatal(e) =>
```

Comment:
if you use the scala Try rather than the try, it handles the non fatal stuff for you

Reply:
also I'm pretty sure it has a toEither as well